projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -460,6 +463,69 @@ public function add_gutenberg_menus() {
 		$this->update_menu( 'gutenberg-edit-site', 'https://wordpress.com/site-editor/' . $this->domain, null, null, null, 59 );
 	}
 
+	/**
+	 * Add the calypso /woocommerce-installation/ menu item.
+	 */
+	public function add_woocommerce_installation_menu() {
+
+		$is_woa_site          = ( new Host() )->is_woa_site();
+		$is_simple_site       = defined( 'IS_WPCOM' ) && IS_WPCOM;
```

Comment:
I suppose instead of checking for WoA vs Simple here in the Admin_Menu class, we could put a method in WPcom_Admin_Menu and Atomic_Admin_Menu, and then put the shared logic in this Admin_Menu class and reference it from the other 2 classes. 🤔

Reply:
Removed this logic and simply reference the base class in `reregister_menu_items()` like `$this->add_woocommerce_installation_menu();` from `class-atomic-admin-menu.php` and `class-wpcom-admin-menu.php`.